src/System.Private.CoreLib/shared/System/DateTime.cs

```diff
@@ -641,32 +641,38 @@ public int CompareTo(DateTime value)
 
         // Returns the tick count corresponding to the given year, month, and day.
         // Will check the if the parameters are valid.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static long DateToTicks(int year, int month, int day)
         {
-            if (year >= 1 && year <= 9999 && month >= 1 && month <= 12)
+            if (year < 1 || year > 9999 || month < 1 || month > 12 || day < 1)
             {
-                int[] days = IsLeapYear(year) ? s_daysToMonth366 : s_daysToMonth365;
-                if (day >= 1 && day <= days[month] - days[month - 1])
-                {
-                    int y = year - 1;
-                    int n = y * 365 + y / 4 - y / 100 + y / 400 + days[month - 1] + day - 1;
-                    return n * TicksPerDay;
-                }
+                ThrowHelper.ThrowArgumentOutOfRange_BadYearMonthDay();
+            }
+            
+            int[] days = IsLeapYear(year) ? s_daysToMonth366 : s_daysToMonth365;
+            if (day > days[month] - days[month - 1])
+            {
+                ThrowHelper.ThrowArgumentOutOfRange_BadYearMonthDay();
             }
-            throw new ArgumentOutOfRangeException(null, SR.ArgumentOutOfRange_BadYearMonthDay);
+
+            int y = year - 1;
+            int n = y * 365 + y / 4 - y / 100 + y / 400 + days[month - 1] + day - 1;
```

Comment:
Btw, this method accesses `static int[]` field 3 times.
I think we can add 
```csharp
static ReadOnlySpan<byte> daysPerMonth366 => new byte[] { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
static ReadOnlySpan<byte> daysPerMonth365 => new byte[] { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
```
to quickly access days per specific month without calculations

Reply:
> this method accesses static int[] field 3 times

it creates a local variable to avoid that so we have only 1 static field access

> to quickly access days per specific month without calculations

I took a look at this and other methods which are using these arrays are relying on the current content. `long ticks` is the number of 100-nanosecond intervals since January 1, 1601 and the number of days since the beginning of the year is needed more frequently